test/emqx_broker_SUITE.erl

```diff
@@ -309,6 +309,26 @@ t_connack_auth_error(Config) when is_list(Config) ->
     ?assertEqual(2, emqx_metrics:val('packets.connack.auth_error')),
     ok.
 
+t_handle_in_empty_client_subscribe_hook({init, Config}) ->
+    Config;
+t_handle_in_empty_client_subscribe_hook({'end', _Config}) ->
+    ok;
+t_handle_in_empty_client_subscribe_hook(Config) when is_list(Config) ->
+    Hook = fun(_ClientInfo, _Username, TopicFilter) ->
+                   EmptyFilters = [{T, Opts#{delete => true}} || {T, Opts} <- TopicFilter],
+                   {stop, EmptyFilters}
+           end,
+    ok = emqx:hook('client.subscribe', Hook, []),
+    try
+        {ok, C} = emqtt:start_link(),
+        {ok, _} = emqtt:connect(C),
+        {ok, _, RCs} = emqtt:subscribe(C, <<"t">>),
+        ?assertEqual([], RCs),
```

Comment:
I see, it seems like the correct thing to do is to add a failure reason in this case.

I'll open a PR that adds an error RC rather than just removing the topic.